modules/exploits/linux/http/panos_management_unauth_rce.rb

```diff
@@ -0,0 +1,197 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Palo Alto Networks PAN-OS Management Interface Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits an authentication bypass vulnerability (CVE-2024-0012) and a command injection
+          vulnerability (CVE-2024-9474) in the PAN-OS management web interface. An unauthenticated attacker can
+          execute arbitrary code with root privileges.
+
+          The following version are affected:
+          * PAN-OS 11.2 (up to and including 11.2.4-h1)
+          * PAN-OS 11.1 (up to and including 11.1.5-h1)
+          * PAN-OS 11.0 (up to and including 11.0.6-h1)
+          * PAN-OS 10.2 (up to and including 10.2.12-h2)
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'watchTowr', # Technical Analysis
+          'sfewer-r7' # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2024-0012'],
+          ['CVE', '2024-9474'],
+          # Vendor Advisories
+          ['URL', 'https://security.paloaltonetworks.com/CVE-2024-0012'],
+          ['URL', 'https://security.paloaltonetworks.com/CVE-2024-9474'],
+          # Technical Analysis
+          ['URL', 'https://labs.watchtowr.com/pots-and-pans-aka-an-sslvpn-palo-alto-pan-os-cve-2024-0012-and-cve-2024-9474/']
+        ],
+        'DisclosureDate' => '2024-11-18',
+        'Platform' => [ 'linux', 'unix' ],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => true, # Executes as root on Linux
+        'Targets' => [ [ 'Default', {} ] ],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
```

Comment:
We should be able to remove the default payload here
```suggestion
```
and might include a comment saying what payloads were tested and working

Reply:
addressed via commit [f9b099a](https://github.com/rapid7/metasploit-framework/pull/19663/commits/f9b099a46dce442eb634515f29a837cd3a5b5315)